src/wmrc/validate.py

```diff
@@ -0,0 +1,167 @@
+        .set_index(["data_year", "id", "name"])
+    )
+
+    return _year_over_year_changes(facility_summary_by_year, current_year)
+
+
+def _year_over_year_changes(metrics_df: pd.DataFrame, current_year: int) -> pd.DataFrame:
+    """Calculate year-over-year changes for columns in a dataframe indexed by both year and another level for entity.
+
+    Args:
+        metrics_df (pd.DataFrame): Data to calculate year-over-year changes for. Must be indexed by year and at least
+            one other level identifying the entity (facility, county, etc). Columns must be numeric metrics we want to
+            compare year over year.
+        current_year (int): The year to compare changes from the previous year.
+
+    Raises:
+        ValueError: If the current year or previous year are in the dataframe's index
+
+    Returns:
+        pd.DataFrame: Each column's pct change, current year value, previous year value, and difference between current and previous year values, indexed by the entity (or whatever was the other level in the input dataframe).
+    """
+
+    previous_year = current_year - 1
+    if current_year not in metrics_df.index:
+        raise ValueError(f"Current year {current_year} not found in index")
+    if previous_year not in metrics_df.index:
+        raise ValueError(f"Previous year {previous_year} not found in index")
+
+    diffs = metrics_df.loc[current_year] - metrics_df.loc[previous_year]
+    pct_change = diffs / metrics_df.loc[previous_year] * 100
+
+    #: have to rename columns after pct_change so that the indices line up properly for the calculation
+    diffs.columns = [f"{col}_diff" for col in diffs.columns]
+    pct_change.columns = [f"{col}_pct_change" for col in pct_change.columns]
+
+    values_current_year = metrics_df.loc[current_year].copy().reindex(index=diffs.index)
+    values_current_year.columns = [f"{col}_{current_year}" for col in values_current_year.columns]
+
+    values_previous_year = metrics_df.loc[previous_year].copy().reindex(index=diffs.index)
+    values_previous_year.columns = [f"{col}_{previous_year}" for col in values_previous_year.columns]
+
+    everything = pd.concat([pct_change, values_current_year, values_previous_year, diffs], axis=1)
+
+    return everything[
+        list(interleave([pct_change.columns, values_current_year.columns, values_previous_year.columns, diffs.columns]))
+    ]
+
+
+def run_validations():
+
+    #: Get records from salesforce, run summary methods
+    wmrc_skid = Skid()
+    records = wmrc_skid._load_salesforce_data()
+    _ = records.deduplicate_records_on_facility_id()
+    facility_summary_df = summarize.facility_metrics(records)
+    county_summary_df = summarize.counties(records)
+
+    #: Calc year-over-year changes
+    facility_changes = facility_year_over_year(facility_summary_df, records.df, 2023)
+    county_changes = county_year_over_year(county_summary_df, 2023)
+    state_changes = state_year_over_year(county_summary_df, 2023)
```

Comment:
Not sure if this is wanted but it makes the script run the previous year without requiring edits to the script?

```suggestion
    from datetime import datetime

    previous_year = datetime.now().year - 1
    facility_changes = facility_year_over_year(facility_summary_df, records.df, previous_year)
    county_changes = county_year_over_year(county_summary_df, previous_year)
    state_changes = state_year_over_year(county_summary_df, previous_year)
```